### PR TITLE
[besu] enable one prometheus for all organizations of a network

### LINKED ISSUE #1416: Setup Prometheus on Kubernetes
### Description
As a Kubernetes operator I want to automatically add Prometheus on the Kubernetes cluster so that I can add monitoring for different Blockchain platforms.

### Acceptance Criteria
1. Add prometheus chart
2. Deploy prometheus from shared playbooks so that all platforms deploy prometheus.
3. Prometheus enablement, passwords must be provided via network.yaml
4. Ensure one Prometheus deployment per cluster

### PR DESCRIPTION
**Pull request by saurabhkumarkardam**
### **Commit to be reviewed**
---

**feat(besu): enable one prometheus for all organizations of a network**

```
This pull request enables one Prometheus server to be used for all organizations in a network.

Changes:
 • The Prometheus code has been migrated to the setup-k8s-environment.yaml playbook in the shared platform.
 • Unwanted variables have been removed from the Prometheus Ansible role.
 • The Prometheus code has been modified to first fetch all the targets from different organizations and then deploy Prometheus in the default namespace with the same fetched targets.
 • Introduced "prometheus-routing.tpl" file to create Ambassador routing rules dynamically for the Prometheus server.

Benefits:
 • This feature reduces the complexity of managing Prometheus by eliminating the need to deploy separate Prometheus servers for each organization.
 • It also improves the efficiency of Prometheus by allowing it to collect metrics from all organizations in a single place.
```

fixes #1416

### CODE REVIEW

File: platforms/shared/configuration/setup-k8s-environment.yaml [hunk 1]
```diff
@@ -100,6 +100,15 @@
     when: 
     - network.type == 'fabric'
     - network.env.type == 'operator'
+
+  # Deploy prometheus server
+  - name: Deploy prometheus server
+    include_role:
+      name: setup/prometheus
+    vars:
+      prometheus_port: "{{ network.organizations[0].prometheus.port }}"
```
Comment (suvajit-sarkar) [inline platforms/shared/configuration/setup-k8s-environment.yaml:109]: better to have Prometheus check and port at network level. and use http/https ports for Prometheus

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]

File: platforms/shared/configuration/roles/setup/prometheus/templates/prometheus_release.tpl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+kube-state-metrics:
+    enabled: false
+prometheus-node-exporter:
+    enabled: false
+prometheus-pushgateway:
+    enabled: false
+server:
+    service:
+        annotations:
+            getambassador.io/config: |
```
[the remark this answers is not in the document]
Comment (saurabhkumarkardam) [reply platforms/shared/configuration/roles/setup/prometheus/templates/prometheus_release.tpl]: Fixed.

File: platforms/shared/configuration/roles/setup/prometheus/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- name: Updating helm values config 
+  template:
+    src={{ playbook_dir }}/../../../platforms/shared/configuration/roles/setup/prometheus/templates/prometheus_release.tpl
+    dest={{ playbook_dir }}/../../../platforms/shared/configuration/build/prometheus.yaml
+  when: targets | length > 0
+
+# Add the Prometheus Helm chart repository if targets exist.
+- name: Add stable chart repo
+  kubernetes.core.helm_repository:
+    name: stable
```
[the remark this answers is not in the document]
Comment (saurabhkumarkardam) [reply platforms/shared/configuration/roles/setup/prometheus/tasks/main.yaml]: Fixed.

File: platforms/shared/configuration/setup-k8s-environment.yaml (hunk 1 above)
Comment (saurabhkumarkardam) [reply platforms/shared/configuration/setup-k8s-environment.yaml]: Fixed.

File: platforms/shared/configuration/setup-k8s-environment.yaml [hunk 4]
```diff
@@ -100,6 +100,17 @@
...
     - network.env.type == 'operator'
+
+  # Deploy prometheus server
+  - name: Deploy prometheus server
+    include_role:
+      name: setup/prometheus
+    vars:
+      prometheus_port: "{{ network.prometheus.port }}"
+    when: 
+    - network.prometheus.enabled == "true"
```
Comment (sownak) [inline platforms/shared/configuration/setup-k8s-environment.yaml:111]: Use "network.prometheus is defined and network.prometheus.enabled "

File: platforms/hyperledger-besu/configuration/samples/network-besu-new-memberorg.yaml [hunk 5]
```diff
@@ -30,6 +30,16 @@ network:
     loadBalancerSourceRanges: # (Optional) Default value is '0.0.0.0/0', this value can be changed to any other IP adres or list (comma-separated without spaces) of IP adresses, this is valid only if proxy='ambassador'
     retry_count: 20                 # Retry count for the checks on Kubernetes cluster
     external_dns: enabled           # Should be enabled if using external-dns for automatic route configuration
+
+  # Prometheus setup section
+  prometheus:
+    enabled: "true"                 # Set to "true" to enable Prometheus monitoring, or "false" to disable it.
```
Comment (sownak) [inline platforms/hyperledger-besu/configuration/samples/network-besu-new-memberorg.yaml:36]: remove double quotes for true so that it acts as boolean

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/shared/configuration/setup-k8s-environment.yaml (hunk 4 above)
Comment (saurabhkumarkardam) [reply platforms/shared/configuration/setup-k8s-environment.yaml]: Fixed.

File: platforms/hyperledger-besu/configuration/samples/network-besu-new-memberorg.yaml (hunk 5 above)
Comment (saurabhkumarkardam) [reply platforms/hyperledger-besu/configuration/samples/network-besu-new-memberorg.yaml]: Fixed.

**Review by sownak — APPROVED** [review]

**Pull request merged**